platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/PkiTools.java

```diff
@@ -260,6 +263,53 @@ public static X500Name convertDistinguishedName(String... tuples) {
     }
 
     /**
+     * Create an X509 general name based on the specified string which supports a format similar to
+     * OpenSSL X509 configuration as: {@code tag:name} where tag can be one of:
+     * <ul>
+     * <li>email - email subject</li>
+     * <li>URI - uniformed resource identifier</li>
+     * <li>RID - registered id</li>
+     * <li>DNS - hostname</li>
+     * <li>IP - ip address (V4 or V6)</li>
+     * <li>dirName - directory name</li>
+     * </ul>
+     *
+     * @param name a string representing an X509 general name in an openssl format
+     * @return the corresponding parsed X509 general name
+     * @throws IllegalArgumentException if the string is <code>null</code>, empty or cannot be parsed properly
+     * @see <a href="https://www.openssl.org/docs/man1.0.2/apps/x509v3_config.html">OpenSSL X509 V3 Configuration</a>
+     * @see <a href="https://en.wikipedia.org/wiki/SubjectAltName">Subject Alternative Names on Wikipedia'</a>
+     */
+    public static GeneralName makeGeneralName(String name) {
+        Validate.isTrue(name != null, "Certificate general name cannot be null");
+        final int i = name.indexOf(':');
+
+        Validate.isTrue(i != -1, "General name components must be in the format tag:value");
+        final String tag = name.substring(0, i)
+                .trim();
+        final String gname = name.substring(i + 1)
+                .trim();
+
+        Validate.isTrue(!gname.isEmpty(), "General name cannot be empty");
+        switch (tag) {
```

Comment:
✏️  How about putting the `tag` names in a `static final Map<String, Integer>` and do?
```
Integer tagNo = tags.get(tag);

if (tagNo == null) {
    throw new ...
}

return new GeneralName(tagNo, gname)
```

If we keep the `switch`, should we define constants for the different tag names?